### PR TITLE
Shift Motion Permissions to Join Room from BalloonShake

### PR DESCRIPTION
**Pull request by juxd**
Device Permission for `DeviceMotionEvent` is now moved to the Join Room page.

### Intended flow:
1. User keys in Room ID and attempts to join.
2. On click, Pinda performs a check on availability of permissions for user:
    - If `DeviceMotionEvent` is available on client's browser: a modal will appear for iOS, but automatically seeks for Android
    - Based on this result, permission would be set to `GRANTED` or `DENIED`.
    - Otherwise, permission is automatically set to `DENIED`.
    - Settings would be saved based on previous visits in the same session for iOS
3. If permissions is set to `GRANTED`, a room connection attempt would be made.
4. Otherwise, a notice would appear notifying the user of a lack of permissions.

### CODE REVIEW

File: Web/src/components/games/BalloonShake/GamePrep.tsx [hunk 1]
```diff
@@ -45,6 +46,19 @@ const GamePrep: React.FC<IProps> = ({
             Start Game!
           </BigButton>
         )}
+    {permission === MotionPermission.DENIED
+        && (
+          <>
+            <ShockedPindaSVG />
+            <h1>
+              Uh Oh!
```
Comment (indocomsoft) [inline Web/src/components/games/BalloonShake/GamePrep.tsx]:
```suggestion
              Oopsie Woopsie!
```

File: Web/src/components/games/BalloonShake/GamePrep.tsx [hunk 2]
```diff
@@ -45,6 +46,19 @@ const GamePrep: React.FC<IProps> = ({
...
+            </h1>
+            <p>
+              Permissions to your device&apos;s motion sensor was not granted, so
+              you cannot play this game.
```
Comment (indocomsoft) [inline Web/src/components/games/BalloonShake/GamePrep.tsx]: Are we gonna give them the instructions on how to reset such that they get the prompt again?

File: Web/src/components/games/BalloonShake/GamePrep.tsx (hunk 1 above)
Comment (hzxcaryn) [reply Web/src/components/games/BalloonShake/GamePrep.tsx]: LOL

File: Web/src/components/games/BalloonShake/GamePrep.tsx (hunk 2 above)
Comment (hzxcaryn) [inline Web/src/components/games/BalloonShake/GamePrep.tsx]:
```suggestion
              you can&apos;t play this game.
```

Comment (indocomsoft) [reply Web/src/components/games/BalloonShake/GamePrep.tsx]: So formal much wow

File: Web/src/components/games/BalloonShake/GamePrep.tsx (hunk 1 above)
Comment (indocomsoft) [reply Web/src/components/games/BalloonShake/GamePrep.tsx]: Lol jk

Comment (juxd) [reply Web/src/components/games/BalloonShake/GamePrep.tsx]: sorry made a fucky wucky >w<

File: Web/src/components/games/BalloonShake/GamePrep.tsx (hunk 2 above)
Comment (indocomsoft) [reply Web/src/components/games/BalloonShake/GamePrep.tsx]: From @juxd : It's unlikely that user will even enter this state

**Pull request merged**